src/panels/alarm-table/hooks/useAlarmTableMenu.ts

```diff
@@ -19,16 +19,45 @@ export const useAlarmTableMenu = (rowClicked) => {
         headerGroup?.classList.add('table-body')
     }
 
+    const processRowClicked = (e: Event) => {
+        const index = (e.currentTarget as Element).getAttribute('data-rowindex')
+
+        if (index !== null && index?.length > 0) {
+          rowClicked(parseInt(index, 10), e as MouseEvent, false)
+        }
+    }
+
+    const processContextMenu = (e: Event) => {
+        const index = (e.currentTarget as Element).getAttribute('data-rowindex')
+
+        if (index !== null && index?.length > 0) {
+          contextMenu(parseInt(index, 10), e as MouseEvent)
+        }
+    }
+
     useEffect(() => {
-        addClassToTableBody();
+        addClassToTableBody()
+
         const rows = table.current?.querySelectorAll('.table-body div[role="row"]')
+
+        const onRowClicked = (e: Event) => processRowClicked(e)
+        const onContextMenu = (e: Event) => processContextMenu(e)
+
         rows?.forEach((row, index) => {
-            row.removeEventListener('click', (e: unknown) => rowClicked(index, e as MouseEvent))
-            row.removeEventListener('contextmenu', (e: unknown) => contextMenu(index, e as MouseEvent))
-            row.addEventListener('click', (e: unknown) => rowClicked(index, e as MouseEvent))
-            row.addEventListener('contextmenu', (e: unknown) => contextMenu(index, e as MouseEvent))
+            row.setAttribute('data-rowindex', '' + index)
+            row.addEventListener('click', onRowClicked)
```

Comment:
To fix this permanently, we'd just have to attach the event listener to an element that isn't removed from the DOM (a parent), or add a few more variable changes to the useEffect callback. You mentioned in the PR description that when you change the date range, the listeners aren't re-attached, so we might just have to pull in the date values into the [table, menu, menuOpen, rowClicked] variables.